From 746a4217713d165fccf2563d1917bd65761102f1 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Mon, 24 Feb 2014 13:46:59 -0800
Subject: [PATCH 496/974] MIPS: OCTEON: Nand: Fix setting the ecc_strength.

Set the ecc_strength only if it is not set.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/cavium-octeon/octeon-nand.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/mips/cavium-octeon/octeon-nand.c b/arch/mips/cavium-octeon/octeon-nand.c
index 8a8db46..6a9766b 100644
--- a/arch/mips/cavium-octeon/octeon-nand.c
+++ b/arch/mips/cavium-octeon/octeon-nand.c
@@ -923,8 +923,9 @@ static int octeon_nand_scan_onfi(struct octeon_nand *priv)
 		 * OCTEON requires ((15 * t) + 7) / 8
 		 */
 		priv->nand.ecc.bytes = ((15 * priv->nand.ecc.strength) + 7) / 8;
-		priv->nand.ecc.strength =
-			(priv->nand.ecc.bytes * 8) / fls(8 * priv->nand.ecc.size);
+		if (!priv->nand.ecc.strength)
+			priv->nand.ecc.strength =
+				(priv->nand.ecc.bytes * 8) / fls(8 * priv->nand.ecc.size);
 		priv->nand.ecc.steps = (1 << priv->nand.page_shift) / priv->nand.ecc.size;
 		priv->nand.ecc.calculate = octeon_nand_bch_calculate;
 		priv->nand.ecc.correct = octeon_nand_bch_correct;
-- 
2.6.2

